File: Examples/VisualBasic/WorkingWithResources/GetResourceOvertime.vb

Namespace WorkingWithResources
    Public Class GetResourceOvertime
        Public Shared Sub Run()
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType)

            ' ExStart:GetResourceOvertime
            ' Create project instance
            Dim project1 As New Project(dataDir + "project_test.mpp")

            ' Display overtime related parameters for all resources
            For Each res As Resource In project1.Resources
                If res.[Get](Rsc.Name) IsNot Nothing Then
                    Console.WriteLine(res.[Get](Rsc.OvertimeCost))
                    Console.WriteLine(res.[Get](Rsc.OvertimeWork).ToString())
                    Console.WriteLine(res.[Get](Rsc.OvertimeRateFormat).ToString())
                End If
            Next
            ' ExEnd:GetResourceOvertime
        End Sub
    End Class
End Namespace

File: Examples/CSharp/WorkingWithResources/GetResourceOvertime.cs
namespace Aspose.Tasks.Examples.CSharp.WorkingWithResources
{
    using System;

    internal class GetResourceOvertime
    {
        public static void Run()
        {
            // The path to the documents directory.
            var dataDir = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType.FullName);

            //ExStart:GetResourceOvertime
            //ExFor: Rsc.OvertimeCost
            //ExFor: Rsc.OvertimeWork
            //ExFor: Rsc.OvertimeRateFormat
            //ExSummary: Shows how to read resource overtime values.
            // Create project instance
            var project = new Project(dataDir + "ResourceOvertime.mpp");

            // Display overtime related parameters for all resources
            foreach (var res in project.Resources)
            {
                if (res.Get(Rsc.Name) == null)
                {
                    continue;
                }

                Console.WriteLine(res.Get(Rsc.OvertimeCost));
                Console.WriteLine(res.Get(Rsc.OvertimeWork).ToString());
                Console.WriteLine(res.Get(Rsc.OvertimeRateFormat).ToString());
            }
            //ExEnd:GetResourceOvertime
        }
    }
}
